Apollo: replace webapollo.py by python-apollo

abretaud opened the pull request:
The very beginning of replacing webapollo.py by python-apollo, to remind me that I should work on it some day!

erasche commented:
The downside of this is that the tool environments are not available when the codefile is executed.

So the admin will have to install python-apollo into the galaxy venv so it can be used in the codefile execution

abretaud commented:
I think we can keep a minimal webapollo.py for dynamic options, but I'm more thinking of making the tools commands use python-apollo instead of webapollo.py
There is a lot of code in webapollo.py that is not needed for dynamic options, and that is redundant with python-apollo (and probably less up to date)

erasche commented:
Oh! gotcha, thought for everything but for the tools, nah makes sense :)

abretaud commented:
Friday commit:
- moved most of the webapollo.py code to python-apollo, kept the code used by dynamic_options
- added tests

it depends on https://github.com/galaxy-genome-annotation/python-apollo/pull/17
and I haven't finished to test all tools
to be continued next week!

erasche commented on tools/apollo/create_or_update_organism.py:
```diff
@@ -36,7 +36,7 @@ def IsOrgCNSuffixEnabled():
         return False
     value = os.environ['GALAXY_APOLLO_ORG_SUFFIX']
-    if value.lower() in ('true', 't', '1'):
+    if value.lower() in ('id', 'email'):
```
you cast to lower here, but not in the later check. Maybe just `return "id"` and `return "email"` here and use the output of this to check which branch below?

abretaud commented:
It's still red, but it *could* become green soon, but first we need to:
- get a few bugs fixed in Apollo (https://github.com/galaxy-genome-annotation/python-apollo/pull/18)
- then merge https://github.com/galaxy-genome-annotation/python-apollo/pull/17 + make a new release
- have it in bioconda
- bump versions here

![easy](https://media0.giphy.com/media/3o7btNa0RUYa5E7iiQ/giphy.gif)

erasche commented:
🎉🌟✨🎊🏆🍻

abretaud commented:
OMG IT4S GREEN //§!
But: maybe we should wait for Apollo 2.4.2 before merging...
Anyway, ready for review

(fixes #46 by the way)

abretaud commented:
OMG IT4S GREEN AGAIN //§!
And Apollo 2.5.0 is out too, so I think it's ready
@erasche @bgruening do you have time (and want;)) to review it?
I *think* it should work with older versions of apollo, but the best would probably be to update usegalaxy.eu/apollo/, do you think it would be possible?

bgruening commented on tools/apollo/list_organisms.xml:
```diff
@@ -6,10 +6,12 @@
-@ADMIN_AUTH@
-$__user_email__
-> $output]]></command>
+  which python &&
```
this can be removed I assume?

bgruening commented on tools/apollo/export.py:
```diff
@@ -3,89 +3,86 @@
+            raise Exception("Apollo failed to prepare the file for download: %s" % uuid_gff)
+        args.gff.write(wa.io.download(uuid_gff['uuid'], output_format="text"))
+
+        time.sleep(1)
```
Should we use flush() here instead of the time?

abretaud replied:
I think the sleep calls are there to avoid timing problems with the apollo api, not for file writing

erasche commented:
> do you think it would be possible

Since I'm in :beach_umbrella:... here's how to update it if you have time @abretaud 

- https://build.galaxyproject.eu/job/usegalaxy-eu/job/apollo-builder-eu/ runs to build an EU specific version.
  - It automatically runs 1/week I think? It uses `application.properties` to tell if there's a new version of apollo, and then...
  - it uploads new apollo versions to here: https://usegalaxy.eu/static/vgcn/
- the [infra playbook variable here](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/group_vars/apollo.yml#L30) controls the apollo version that's deployed (it's just a substring of the .war that's on static), whenever there's a new version on the static file repo published, just make a PR to update that variable and it will be done in 24h :)

**tl;dr**: ensure new version is available + PR update to the variable

abretaud commented:
@erasche nice! I'll try that thanks!

abretaud commented:
The update went well apparently \o/

erasche commented:
oh thank goodness lol :pray:
